### PR TITLE
feature: export by project

### PR DESCRIPTION
**Pull request by jsachs**

### CODE REVIEW

**Comment by mmccall** [comment]
image

I think your calculations are off in your export for by project. It is telling us the percentage of my time allocated to that project in the attached screenshot, not the percentage of the projects time that my allocation constitutes.  Basically, the % should be '1' I'd think on the by project breakout, right?  If another person billed to project 1, that value with decrement proportionally.

File: api/controllers/csv.js [hunk 1]
```diff
@@ -52,6 +52,56 @@ exports.getReportPastWeekCSV = function (req, res, next) {
...
+            });
+        });
+
+        ret.sort((function(i) {
+            return function(a, b) {
+                return (a[i] === b[i] ? 0 : (a[i] < b[i] ? -1 : 1));
+            };
+        })(0));
+
+        res.set('Content-Disposition', 'attachment; filename="project.csv"');
```
Comment (mmccall) [inline api/controllers/csv.js]: re: your comment on my PR, I'll align my routes to reflect yours.  Can you change the .csv export name on person to align with this convention though? (line 103)

**Comment by jsachs** [comment]
@mmccall I'm pretty sure this is what Hallie wants. Here is the "by project" spreadsheet:

image

% of Hours is greater than 100%, showing that it is by person.

**Comment by jsachs** [comment]
@mmccall looks like you left some merge lines in.

**Comment by mmccall** [comment]
@jsachs just tried to use the web ui to reconcile it, which was a mistake.  Missed a bracket.

**Pull request merged**